client/dashboard/domains/domain-connection-setup/components/dns-records-table.tsx

```diff
@@ -0,0 +1,230 @@
+import { useSuspenseQuery } from '@tanstack/react-query';
+import { DataViews } from '@wordpress/dataviews';
+import { __ } from '@wordpress/i18n';
+import { useMemo } from 'react';
+import { DataViewsCard } from '../../../components/dataviews-card';
+import type { Field, ViewTable } from '@wordpress/dataviews';
+
+import './dns-records-table-style.scss';
+
+const viewSuggested: ViewTable = {
+	type: 'table',
+	page: 1,
+	fields: [ 'currentValue', 'expectedValue', 'status' ],
+	layout: {
+		enableMoving: false,
+	},
+};
+
+const viewAdvanced: ViewTable = {
+	type: 'table',
+	page: 1,
+	fields: [ 'type', 'name', 'currentValue', 'expectedValue', 'status' ],
+	layout: {
+		enableMoving: false,
+	},
+};
+
+interface DnsRecordSuggestedVerification {
+	currentValue: string;
+	expectedValue: string;
+	status: React.ReactNode;
+}
+
+interface DnsRecordAdvancedVerification {
+	type: string;
+	name: string;
+	currentValue: string;
+	expectedValue: string;
+	status: React.ReactNode;
+}
+
+const fieldsSuggested: Field< DnsRecordSuggestedVerification >[] = [
+	{
+		id: 'currentValue',
+		label: __( 'Current Value' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+	{
+		id: 'expectedValue',
+		label: __( 'Expected Value' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+	{
+		id: 'status',
+		label: __( 'Status' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+];
+
+const fieldsAdvanced: Field< DnsRecordAdvancedVerification >[] = [
+	{
+		id: 'type',
+		label: __( 'Type' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+	{
+		id: 'name',
+		label: __( 'Name' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+	{
+		id: 'currentValue',
+		label: __( 'Current Value' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+	{
+		id: 'expectedValue',
+		label: __( 'Expected Value' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+	{
+		id: 'status',
+		label: __( 'Status' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+];
+
+const VerifiedBadge = () => {
+	return <Badge intent="success">{ __( 'Verified' ) }</Badge>;
+};
+
+const VerifyingBadge = () => {
+	return <Badge intent="warning">{ __( 'Verifying' ) }</Badge>;
+};
+
+const aRecordData = ( currentValue: string, expectedValue: string ) => {
+	return {
+		type: 'A',
+		name: '@',
+		currentValue: currentValue,
+		expectedValue: expectedValue,
+		status: currentValue === expectedValue ? <VerifiedBadge /> : <VerifyingBadge />,
```

Comment:
It would be better to define the `getValue` function and `render` function for the `status` field

```js
const fields = [
	{
		id: 'status',
		label: __( 'Status' ),
		readOnly: true,
		enableHiding: false,
		enableSorting: false,
		filterBy: false,
		getValue: ( { item } ) => item.currentValue === item.expectedValue,
		render: ( { field, item } ) => <VerificationBadge isVerified={ field.getValue( { item } ) } />,
	}
]
```

Reply:
Besides, is it possible both values are undefined (i.e., `-`) ?